runner/src/server/plugins/engine/services/addressService.ts

```diff
@@ -11,7 +12,7 @@ export async function findByPostcode(key, postcode) {
   const { payload, error } = await getJson(findByPostcodeUrl);
 
   if (error) {
-    console.error(error);
+    logger.error(error);
     return [];
   }
```

Comment:
you're using pino logger, but not the one that's been registered to the server. Take a look at the constructor for `runner/src/server/services/webhookService.ts` if you want it to work in a service

Reply:
Thanks, I understand now.

Before I do this, can I double check the actual approach you have in mind?

We could either pass the server.logger in to the function itself as an extra parameter, or we could convert the addressService to a class and pass the logger in the constructor?

Either way, we don't appear to actually use this addressService at all right now so we won't actually be passing it in from anywhere, but worth us getting it right now for when it is in-use.

Reply:
I think convert it to a class, so we can register and access the service from anywhere within the server. When we register services to the server, using schmervice, schmervice will initialise the object with `(server, name)`. 

so your constructor will look like this
```.ts
class MyService {
  constructor(server, name) { // or just (server) 
    this.logger = server.logger 
  }
}
```

or you can just delete it, since neither depts are using it :p

Reply:
Thanks, I think I'll try to amend it rather than remove it.

How do you feel about moving the service into the src/server/services folder so it's with the rest of the services (eg. WebhookService). It'll look cleaner when we import and register the service, but I'm unsure if there's a reason it currently lives in the engine.

Reply:
Yep, that sounds good! I didn't realise it wasn't already..! 😱